client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -176,10 +176,21 @@ export function SitesDashboard( {
 		ref: 'topbar',
 	} );
 	const { __, _n } = useI18n();
-	const { data: allSites = [], isLoading } = useSiteExcerptsQuery(
+	const { data: liveSites = [], isLoading } = useSiteExcerptsQuery(
 		[],
 		( site ) => ! site.options?.is_domain_only
 	);
+
+	const { data: deletedSites = [] } = useSiteExcerptsQuery(
+		[],
+		( site ) => ! site.options?.is_domain_only,
+		'deleted'
+	);
+
+	deletedSites.forEach( ( site ) => {
+		site.is_deleted = true;
+	} );
```

Comment:
We shouldn't make this change during render. [We could probably use the select property from react-query](https://tkdodo.eu/blog/react-query-data-transformations).